### PR TITLE
Syncs post endpoint v1.0

### PR DESCRIPTION
**Pull request by zinigor**
@vindl  please take a look at this one, because I have changed the way we check the meta key here. When using the object notation I would get `trying to get a property of non-object`.

### CODE REVIEW

**Comment by zinigor** [comment]
Master issue: #7026

File: json-endpoints/class.wpcom-json-api-post-endpoint.php [hunk 1]
```diff
@@ -65,35 +65,6 @@ function __construct( $args ) {
...
-		 * Filters the meta keys accessible by the REST API.
-		 * @see https://developer.wordpress.com/2013/04/26/custom-post-type-and-metadata-support-in-the-rest-api/
-		 *
-		 * @module json-api
-		 *
-		 * @since 2.2.3
-		 *
-		 * @param array $whitelisted_meta Array of metadata that is accessible by the REST API.
-		 */
- 		if ( in_array( $key, apply_filters( 'rest_api_allowed_public_metadata', $whitelisted_meta ) ) )
```
Comment (jeherve) [inline json-endpoints/class.wpcom-json-api-post-endpoint.php]: We use that filter in a few places in Jetpack. If it's not used anymore, we should remove it from there, and add the filter to the list of deprecated hooks in `class.jetpack.php`.

**Comment by eliorivero** [comment]
captura de pantalla 2017-04-24 a las 15 58 50

When I switch to this branch, I get this in Calypso when I try to edit a post.

File: json-endpoints/class.wpcom-json-api-post-endpoint.php (hunk 1 above)
Comment (zinigor) [inline json-endpoints/class.wpcom-json-api-post-endpoint.php]: According to #7099 there are plans to sync metadata, so we will still need that filter in the future, let's not remove it from all other places.

**Pull request merged**